### ISSUE TITLE
Reuse existing image file

### ISSUE THREAD

**Issue by jcoliz** [ShapeCrawler/ShapeCrawler#809]
`AddPicture()` duplicates the image source.

Use test [AddPicture_should_not_duplicate_the_image_source_When_the_same_image_is_added_twice](https://github.com/ShapeCrawler/ShapeCrawler/blob/2f4cb91efe059a7ca695d2440339f1510f2ad1e8/test/ShapeCrawler.Tests.Unit/ShapeCollectionTests.cs#L571-L588) to reproduce this bug:
```C#
[Test]
public void AddPicture_should_not_duplicate_the_image_source_When_the_same_image_is_added_twice()
{
    // Arrange
    var pres = new Presentation(StreamOf("008.pptx"));
    var shapes = pres.Slide(1).Shapes;
    var image = StreamOf("png image-1.png");
    
    // Act
    shapes.AddPicture(image);
    shapes.AddPicture(image);
    
    // Assert
    pres.SaveAs("output.pptx");
    // Check the folder output.pptx/ppt/media. This folder should contain only one image file.
    // TODO: Add assertion
}
```

---

### Initial Message

Doing some research into why my PPTX files are 30MB, and I discovered...

While running Powerpoint, if I add the same image into the document several times, the image is only added once to the file's `media` directory, and all shapes which use it refer to the same part ID in their blip.

However, using ShapeCrawler, every call to `AddPicture` adds the image _again_.

It would be great to have some mechanism where I could signify reuse. Or even better, if it happened automatically. So that the same image added twice would look up the existing part and refer to that instead of embedding a whole new one. I'd be happy to specify a string file name to use as a key.

**Comment by blockingHD** [comment]
Having played around with this in PowerPoint it is comparing file contents not just the file name. 
Is there a built in way of doing these comparisons? 
If not I guess a signature will need to be generated for each file and compared. 

Along with this how do we access the contents of the media file as the IPackageFeature interface is internal?

**Comment by ashahabov** [comment]
> If not I guess a signature will need to be generated for each file and compared.

Correct. We can just use some any hash function for that.

> Along with this how do we access the contents of the media file as the IPackageFeature interface is internal?

I think the logic "Add or Re-Use existing source image" should be handled in the method [SlideShapes.CreatePPicture()](https://github.com/ShapeCrawler/ShapeCrawler/blob/52b2b8e585a567cd45c87c4fefd8ae00020dee7c/src/ShapeCrawler/ShapeCollection/SlideShapes.cs#L567) by managing `ImagePart`.

**Comment by jcoliz** [comment]
@blockingHD Right, PowerPoint is smart enough to de-dupe based on file contents. In my description above, I was suggesting that we could send in a file key (e.g. file name) as a sort of crutch. 

As a library user, I don't absolutely NEED ShapeCrawler to work out that they're the same image. It would be cool. Still, I'll know at the time of adding. So if it's easier, my suggestion is giving a key. But yeah taking the extra step of hashing the contents would be better.

**Comment by jcoliz** [comment]
Put up a first crack at solving this. Actually looks pretty straightforward.

@ashahabov Any chance I could convince you to use `Explicit` instead of `Ignore` on failing/incomplete tests? Explicit still keeps them out of automated test runs, but still allows to explicitly run/debug the test in VS Code or run the test via `dotnet test --filter {testname}`

**Comment by FabrisTB** [comment]
Do you know if it's possible to use the same image on different slides ? Your solution works but only if it's on the same slide.

**Comment by jcoliz** [comment]
@FabrisTB It's a work in progress. De-duping across multiple slides is the next thing to figure out.

**Comment by ashahabov** [comment]
> @ashahabov Any chance I could convince you to use Explicit instead of Ignore on failing/incomplete tests? Explicit still keeps them out of automated test runs, but still allows to explicitly run/debug the test in VS Code or run the test via dotnet test --filter {testname}

@jcoliz, yes, I agree to further use the attribute `Explicit` for such tests.
